### PR TITLE
Fix unamange cluster API needing cluster_id

### LINKED ISSUE #365: Unmanage API requires unnecessary parameter cluster_id
URI for cluster unmanage API call looks like: `hostname/api/1.0/{imported_cluster_id}/unmanage` and it requires to send data in format: `{"cluster_id":"{imported_cluster_id}"}`. If the json with these data is not provided then the call is not processed:
```
$ curl -XPOST -H "Authorization: Bearer access_token" -H "Content-Type: application/json" hostname/api/1.0/clusters/{imported_cluster_id}/unmanage
{"errors":{"message":"Invalid JSON received."}}
```
but when the `cluster_id` parameter is provided, it works:
```
$ curl -XPOST -H "Authorization: Bearer access_token" -H "Content-Type: application/json" -d '{"cluster_id":"{imported_cluster_id}"}' hostname/api/1.0/clusters/{imported_cluster_id}/unmanage
{"job_id":"53474cf0-72d1-4edc-854f-a63e91e17956"}
```
User should not be required to enter `cluster_id` in json data that are sent with API request as this information is already part of the URI of the API call.

Tested with:
```
tendrl-ansible-1.5.5-1.el7.centos.noarch
tendrl-api-1.6.0-1.el7.centos.noarch
tendrl-api-httpd-1.6.0-1.el7.centos.noarch
tendrl-commons-1.6.0-1.el7.centos.noarch
tendrl-grafana-plugins-1.6.0-1.el7.centos.noarch
tendrl-grafana-selinux-1.5.4-2.el7.centos.noarch
tendrl-monitoring-integration-1.6.0-1.el7.centos.noarch
tendrl-node-agent-1.6.0-1.el7.centos.noarch
tendrl-notifier-1.6.0-1.el7.centos.noarch
tendrl-selinux-1.5.4-2.el7.centos.noarch
tendrl-ui-1.6.0-1.el7.centos.noarch
```

### PR DESCRIPTION
**Pull request by shirshendu**
Also, add spec for unmanage cluster

Fixes #365

### CODE REVIEW

File: lib/tendrl/flow.rb [hunk 1]
```diff
@@ -15,8 +15,8 @@ def initialize(namespace, flow_name, object = nil)
       @flow_name = flow_name
       @object = object
       @objects = @instance[namespace]['objects']
-      @flow = @instance[namespace]['flows'][flow_name] ||
-              @instance[namespace]['objects'][object]['flows'][flow_name]
+      @flow = @instance[namespace]['flows'][flow_name]
+      @flow ||= @objects[object] ? @objects[object]['flows'][flow_name] : nil
```
[the remark this answers is not in the document]
Comment (shirshendu) [reply lib/tendrl/flow.rb]: Ah nevermind, this need not be here. Was just a bit of an extra check to be sure the object exists in our definitions; that I added while fixing the spec.

File: spec/controllers/clusters_controller_spec.rb [hunk 2]
```diff
@@ -48,6 +48,24 @@
...
+      stub_managed_cluster
+      stub_job_creation
+      Tendrl.cluster_definitions = YAML.load_file(
+        'spec/fixtures/definitions/gluster.yaml'
+      )
+    end
+
+    it 'managed' do
+      post '/clusters/6b4b84e0-17b3-4543-af9f-e42000c52bfc/unmanage',
+      '{}', http_env
```
[the remark this answers is not in the document]
Comment (shirshendu) [reply spec/controllers/clusters_controller_spec.rb]: Agreed. Doing.

File: lib/tendrl/flow.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (r0h4n) [reply lib/tendrl/flow.rb]: No dont remove it, we do want the tendrl-api to strictly enforce the definitions , tendrl-api should ideally fetch latest definitions on service startup from etcd "/_NS/node_agent/compiled_definitions/data"

And it should only work for objects/flows/atoms/attributes that are in the definitions yaml

File: spec/fixtures/definitions/tendrl_definitions_node_agent.json [hunk 3]
```diff
@@ -1 +1 @@
...
...size:\n          help: size of the disk\n          type: String\n        size_bios_edd:\n          help: size of bios edd\n          type: String\n        sysfs_busid:\n          help: sysfs bus id\n          type: String\n        sysfs_device_link:\n          help: sysfs device link\n          type: String\n        sysfs_id:\n          help: sysfs id\n          type: String\n        vendor:\n          help: disk vendor name\n          type: String\n      enabled: true\n      help: VirtualDisk\n      list: nodes/$NodeContext.node_id/LocalStorage/Virtio\n      value: nodes/$NodeContext.node_id/LocalStorage/Virtio\n  supported_sds:\n  - ceph\n  - gluster\n  tags:\n    ceph-installer: provisioner/ceph\n    ceph-mon: ceph/mon\n    ceph-osd: ceph/osd\n    ceph-provisioner: provisioner/ceph\n    etcd: tendrl/central-store\n    gluster-provisioner: provisioner/gluster\n    glusterd: gluster/server\n    tendrl-api: tendrl/server\n    tendrl-ceph-integration: tendrl/integration/ceph\n    tendrl-gluster-integration: tendrl/integration/gluster\n    tendrl-monitor: tendrl/monitor\n    tendrl-node-agent: tendrl/node\ntendrl_schema_version: 0.3\n","modifiedIndex":6221697,"createdIndex":6221697}}
```
[the remark this answers is not in the document]
Comment (shirshendu) [reply spec/fixtures/definitions/tendrl_definitions_node_agent.json]: Noted, thanks. Still working on getting tendrl and gluster to play nice over etcd.

File: lib/tendrl/flow.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (shirshendu) [reply lib/tendrl/flow.rb]: >No dont remove it, we do want the tendrl-api to strictly enforce the definitions , tendrl-api should ideally fetch latest definitions on service startup from etcd "/_NS/node_agent/compiled_definitions/data"
>And it should only work for objects/flows/atoms/attributes that are in the definitions yaml

This statement already did the above, the conditional is not required. I only added it while debugging.

**Comment by r0h4n** [comment]
@shirshendu @gnehapk @fbalak please verify

**Pull request merged**